setup.py

```diff
@@ -263,25 +263,17 @@ def compile_KLU():
             "nbmake",
         ],
         "pandas": [
-            "pandas>=0.24",
+            "pandas>=2.0.3",
         ],
         "jax": [
             "jax==0.4.8",
             "jaxlib==0.4.7",
         ],
         "odes": ["scikits.odes"],
         "all": [
-            "anytree>=2.4.3",
-            "autograd>=1.2",
-            "pandas>=0.24",
-            "scikit-fem>=0.2.0",
-            "imageio>=2.9.0",
-            "pybtex>=0.24.0",
-            "sympy>=1.8",
-            "bpx",
-            "tqdm",
-            "matplotlib>=2.0",
-            "jupyter",
+            "autograd>=1.6.2",
+            "scikit-fem>=8.1.0",
+            "pybamm[examples,plot,cite,latexify,bpx,tqdm,pandas]"
```

Comment:
This syntax is a pretty cool feature which I got from [this blog from Hynek Schlawack](https://hynek.me/articles/python-recursive-optional-dependencies/ ). It requires `pip` v21+ though, which should be fine for us because `pip` v23.0.1 has been backported to ship with Python 3.8.

Reply:
Also, I removed a duplicate entry of `anytree` from the `all` set of extras dependencies since we now have it in the required dependencies after #3475.